### DISCUSSION TITLE
Setting up the new database

### DISCUSSION THREAD

**Discussion by mislavmatijevic** [foivz/pop-app-backend#4] [Q&A]
Setting up a new database is all but easy. Every decision counts.
I've just wrapped up [creating the initial ER diagram](https://github.com/foivz/pop-app-backend/wiki/ER-Diagram) for the new database.
I would like to discuss some aspects of it.

Some decisions that I struggled with the most is:
- whether to use UUIDs or normal sequential IDs
- whether to store date as DATETIME or TIMESTAMP (decided for DATETIME after reading that TIMESTAMP is just a "dumb" 32bit UNIX timestamp that dies in 2038)
- whether to use 1:1 relationship between products/packages and a hypothetical "item" entity that would enbody shared attributes (this seemed like an overkill that would cost us dearly later on in the actual development when JOINing)

**Comment by antoniokunstek** [comment]
Dear Mislav (@mislavmatijevic),

regarding the set items for discussion:

**Whether to use UUIDs or normal sequential IDs**

- UUIDs look more professional and it will be impossible for users to determine which record (id-1, yeay I am a first user 😂) they are in the database. We talked about this over coffee, maybe it's better to leave the sequential id for now as far as MySQL and the H2 database are concerned.

**Whether to store date as DATETIME or TIMESTAMP (decided for DATETIME after reading that TIMESTAMP is just a "dumb" 32bit UNIX timestamp that dies in 2038)**

- unquestionable DATETIME

**Whether to use 1:1 relationship between products/packages and a hypothetical "item" entity that would enbody shared attributes (this seemed like an overkill that would cost us dearly later on in the actual development when JOINing)**

- I leave this choice up to you, but I will add that sometimes it pays to invest more effort (no pain, no gain), and I believe that this will not be a problem for us.

If you have any remaining questions or items for discussion, do not hesitate to ask.
I want to thank you for all the effort and time spent creating this project. You are really doing a great job. 😍
Best regards!
Antonio Kunštek

**Reply by mislavmatijevic** [reply]
Thanks for amazing response! I'll leave 1:1 relationships out for now. I think the model is going to work as it stands.
